### PR TITLE
Use `oc wait` to wait for condition instead of running oc commands

### PR DESCRIPTION
**Pull request by vrutkovs**
There is no need to repeat oc commands - it has `oc wait` subcommands which pauses until a particular `.status.condition` is set to True.

There is no `oc wait` for events, so this would run `oc get events` in a loop and look for `bootstrap-complete` event instead. Loop is limited to 60 attempts with 10 sec pause so bootstrapping is expected to complete in 10+ minutes.

If the cluster didn't start the script would output:
1. bootstrap.service logs
2. clusterversion operator status
3. operator status
4. List of failing pods

### CODE REVIEW

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/141/

File: 07_deploy_masters.sh [hunk 1]
```diff
@@ -69,9 +69,15 @@ done
 echo "Master nodes up, you can ssh to the following IPs with core@<IP>"
 sudo virsh net-dhcp-leases baremetal
 
+# Wait for nodes to become ready
+NUM_LEASES=$(oc --config ocp/auth/kubeconfig get nodes | wc -l)
```
Comment (hardys) [inline 07_deploy_masters.sh]: We should probably rename the variable to NUM_NODES or something?

**Comment by hardys** [comment]
Nice, thanks for this - I wasn't yet aware of the wait subcommand :)

LGTM but had one nit re variable naming which would be nice to fix.

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/143/

File: 07_deploy_masters.sh (hunk 1 above)
Comment (vrutkovs) [reply 07_deploy_masters.sh]: Done

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/145/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/147/

**Comment by derekhiggins** [comment]
LGTM but can we squash it up a bit before merging, there are a bunch of patches fixing things from previous patches etc.. maybe thats why you kept it as a WIP but just making sure

**Comment by vrutkovs** [comment]
Sure, the commits will be squashed, although it doesn't actually wait for bootstrap to complete now - it breaks when VIP switches to first master. Not sure how to add retries there

Perhaps I should leave just `oc wait` for nodes then?

**Comment by hardys** [comment]
> Perhaps I should leave just `oc wait` for nodes then?

Sounds reasonable - leave the current logic until the check for nodes, as by that point we've waited for the VIP failover to happen, via the ssh/hostname check loop?

FWIW I don't see any problem with adding the CVO wait even if that does make the deploy take longer - as we've seen with the kubevirt stuff unexpected things happen when the cluster isn't fully up, so we may as well wait for it to be fully deployed before running any further scripts?

That said we probably do need to thing about the earliest point that workers can be deployed for the final implementation - I'm hoping we can figure out just adding manifests for all the additional pieces in due course, but for now serializing via scripts seems to be a reasonable way to enable further testing?

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/171/

File: 07_deploy_masters.sh [hunk 2]
```diff
@@ -89,6 +89,12 @@ for i in $(seq 0 2); do
...
 
-# Wait for clusterversion operator to report availability
-oc --config ocp/auth/kubeconfig wait clusterversion/version --for condition=available --timeout=600s
+# Wait for bootstrap to complete
+while read line; do
+  echo $line;
+  if [ "$(echo ${line} | awk '{print $4}')" == "bootstrap-complete" ]; then
+    break
+  fi
+done < <(oc --config ocp/auth/kubeconfig get events -n kube-system --no-headers -w -o wide)
```
Comment (hardys) [inline 07_deploy_masters.sh]: You mentioned on slack needing to retry this, could you just omit the `-w` option and call the get events in a loop with grep?

e.g:

```
while ! oc --config ocp/auth/kubeconfig get events -n kube-system --no-headers -o wide | grep -q bootstrap-complete; do
  sleep 10
done

If we need to tolerate failure we can just || true in the conditional?
```

Comment (vrutkovs) [reply 07_deploy_masters.sh]: Right, lets try that

Comment (vrutkovs) [reply 07_deploy_masters.sh]: That worked. Squashed commits

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/192/

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/193/

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/203/

**Comment by vrutkovs** [comment]
@derekhiggins could the jenkins build timeout be extended? Now the script actually waits for bootstrap to complete, which is rarely less than an hour.

I'll merge that once CI runs would not be false-positive

**Comment by hardys** [comment]
@derekhiggins I'm wondering what to do about the CI test failure - it does take a while in my environment to get the bootstrap-complete, but way less that 60mins what sort of env is CI running on?

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/237/

**Comment by derekhiggins** [comment]
Build ABORTED, see build http://10.8.144.11:8080/job/dev-tools/273/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/282/

**Comment by derekhiggins** [comment]
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/294/

**Pull request merged**